### PR TITLE
Add formspec rendering automated testing

### PR DESCRIPTION
**Pull request by rubenwardy**
This implements unit testing of formspec rendering by using the mainmenu to render formspecs. It adds core.take_screenshot and core.on_step to the main menu API. The former is used to create images of the GUI, and the latter is used to interact with the render step.

The aim of this PR is to make PRs like #8524 and #7970 less risky.

## To do

- [ ] Compare output results
- [ ] Integrate with CI
- [ ] Nice side-by-side output on failure

## How to test

Todo

### CODE REVIEW

**Comment by DS-Minetest** [comment]
`on_step` should rather be something like `register_on_render_step`. `core.run_callbacks` can be used for this if you dofile or require `core.get_builtin_path()..".."..DIR_DELIM..".."..DIR_DELIM.."builtin"..DIR_DELIM.."common"..DIR_DELIM.."game"..DIR_DELIM.."register.lua"`,

File: util/fstest/init.lua [hunk 1]
```diff
@@ -0,0 +1,37 @@
+core.set_clouds(false)
+
+local tests = {}
+
+local function load_tests(path)
+	local list = core.get_dir_list(path, false)
+	for _, filename in ipairs(list) do
+		if filename:sub(#filename - 3, #filename) == ".txt" then
```
Comment (DS-Minetest) [inline util/fstest/init.lua]:
```suggestion
		if filename:sub(-4, -1) == ".txt" then
```

File: util/fstest/init.lua [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+			local name = filename:sub(1, #filename - 4)
```
Comment (DS-Minetest) [inline util/fstest/init.lua]:
```suggestion
			local name = filename:sub(1, -5)
```

**Comment by rubenwardy** [comment]
I think I opened this too early, so closing for now

**Pull request closed without merging**